app/controllers/admin/events_controller.rb

```diff
@@ -0,0 +1,49 @@
+class Admin::EventsController < AdminController
+
+  before_action :load_event, only: [:enable, :disable]
+  before_action :is_creator_enabled?, only: [:enable]
+
+  def index
+    @events = Event.all.order(:start_time)
+  end
+
+  def show
+    @event = Event.find_by(id: params[:id])
+    @discussions = @event.discussions.order(:date, :start_time, :id)
+  end
+
+  def enable
+    respond_to do |format|
+     if @event.update_attribute(:enabled, true)
```

Comment:
rather create a module toggleable and define methods `enable` and `disable` in that.
Then include that module in User, Event and Discussion.
Use those methods accordingly.